lib/symbioticpy/symbiotic/tools/cpachecker.py

```diff
@@ -0,0 +1,251 @@
+            elif line.startswith('For your information: CPAchecker is currently hanging at') and not status and isTimeout:
+                status = 'TIMEOUT'
+
+            elif line.startswith('Verification result: '):
+                line = line[21:].strip()
+                if line.startswith('TRUE'):
+                    newStatus = result.RESULT_TRUE_PROP
+                elif line.startswith('FALSE'):
+                    newStatus = result.RESULT_FALSE_REACH
+                    match = re.match('.* Property violation \(([^:]*)(:.*)?\) found by chosen configuration.*', line)
+                    if match and match.group(1) in ['valid-deref', 'valid-free', 'valid-memtrack', 'no-overflow', 'no-deadlock', 'termination']:
+                        newStatus = result.STR_FALSE + '(' + match.group(1) + ')'
+                else:
+                    newStatus = result.RESULT_UNKNOWN
+
+                if not status:
+                    status = newStatus
+                elif newStatus != result.RESULT_UNKNOWN:
+                    status = "{0} ({1})".format(status, newStatus)
+
+        if (not status or status == result.RESULT_UNKNOWN) and isTimeout and returncode in [15, 143]:
+            # The JVM sets such an returncode if it receives signal 15
+            # (143 is 15+128)
+            status = 'TIMEOUT'
+
+        if not status:
+            status = result.RESULT_ERROR
+        return status
+
+
+    def get_value_from_output(self, lines, identifier):
+        # search for the text in output and get its value,
+        # stop after the first line, that contains the searched text
+        for line in lines:
+            if identifier in line:
+                startPosition = line.find(':') + 1
+                endPosition = line.find('(', startPosition) # bracket maybe not found -> (-1)
+                if (endPosition == -1):
+                    return line[startPosition:].strip()
+                else:
+                    return line[startPosition: endPosition].strip()
+        return None
+
+    def llvm_version(self):
+        """
+        Return required version of LLVM
+        """
+        return '3.9.1'
+
+    def set_environment(self, symbiotic_dir, opts):
+        """
+        Set environment for the tool
+        """
+        pass
+
+
+    def preprocess_llvm(self, infile):
+        """
+        A tool's specific preprocessing steps for llvm file
+        before verification itself. Returns a pair (cmd, outputfile),
+        where cmd is the list suitable to pass to Popen and outputfile
+        is the resulting file from the preprocessing
+        """
+        return (None, None)
+
+    def prepare(self):
+        """
+        Prepare the bitcode for verification - return a list of
+        LLVM passes that should be run on the code
+        """
+        """passes = []
+        if not self._options.explicit_symbolic:
+            passes.append('-initialize-uninitialized')
+
+        return passes"""
```

Comment:
This comment was a leftover of my implementation and should be removed - sorry for that